### ISSUE TITLE
Misuse of MAX_PATH

### ISSUE THREAD

**Issue by VideoPlayerCode** [enthought/pywin32-ctypes#91]
Examples:

https://github.com/enthought/pywin32-ctypes/blob/f27d6a0000c611dfaef658ec6dbb9ea905d11deb/win32ctypes/core/ctypes/_system_information.py#L16-L19

https://github.com/enthought/pywin32-ctypes/blob/f27d6a0000c611dfaef658ec6dbb9ea905d11deb/win32ctypes/core/cffi/_system_information.py#L25-L28

Described my Microsoft:

https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation

`In the Windows API (with some exceptions discussed in the following paragraphs), the maximum length for a path is MAX_PATH, which is defined as 260 characters.`

In other words, paths can be 260 characters long. Therefore it is wrong to allocate a buffer that is only big enough to fit 260 characters. We must allocate `MAX_PATH + 1` characters, so that the buffer will fit 260 characters plus the NULL terminator.

PS: "Characters" in Microsoft jargon refers to two-byte Unicode UTF-16 characters. It does not refer to actual bytes.

**Comment by VideoPlayerCode** [comment]
I learned something today. I read the quoted paragraph more closely in MSDN:

`A local path is structured in the following order: drive letter, colon, backslash, name components separated by backslashes, and a terminating null character. For example, the maximum path on drive D is "D:\some 256-character path string<NUL>" where "<NUL>" represents the invisible terminating null character for the current system codepage. (The characters < > are used here for visual clarity and cannot be part of a valid path string.)`

So they're using this as an example: `D:\` (that's 3 characters), and `some 256 character string` and `<NUL>`... That's 3 + 256 + 1 = 260 characters.

Doh, it's safe to allocate MAX_SIZE as the buffer!
